syntax_and_semantics/structs.md

```diff
@@ -11,9 +11,51 @@ struct Point
 end
 ```
 
-The differences between a struct and a class are:
-* Invoking `new` on a struct allocates it on the stack instead of the heap
-* A struct is [passed by value](http://crystal-lang.org/api/Value.html) while a class is passed by reference
+## Differences with a class
+
+### Passing
+
+A struct is [passed by value](http://crystal-lang.org/api/Value.html) while a class is passed by reference.
+
+```crystal
+class Klass
+  property array = ["str"]
+end
+
+struct Strukt
+  property array = ["str"]
+end
+
+def modify(object)
+  object.array << "foo"
+  object.array = ["new"]
+  object.array << "bar"
+end
+
+strukt = Strukt.new
+klass = Klass.new
+modify strukt
+modify klass
+puts strukt.array #=> ["str", "foo"]
```

Comment:
I can't explain why the result isn't `["str", "foo", "bar"]`.

Reply:
A *copy* of the struct gets passed to `modify`. When you do `object.array = ["new"]`, you replace the *reference* to `array` *in the copy* with a new array.

The original reference is only around to receive `"foo"`.

```cr
struct_object.array # Points to Array 0x01
modify(struct_object)

def modify(object)
  object.array  # Copy of struct_object that points to Array 0x01
  object.array << "foo" # Add "foo" to Array 0x01
  object.array = ["new"]
  object.array  # Points to Array 0x02
  object.array << "bar"
end

# Back in main program:
struct_object.array # The original struct_object, still points to Array 0x01
struct_object.array #=> ["str", "foo"]
```

Reply:
Thank you very much @z64 , I'll add your explanation :)